.github/workflows/release.yml

```diff
@@ -39,6 +27,19 @@ jobs:
           email: ${{ env.GIT_EMAIL }}
           token: ${{ env.GITHUB_TOKEN }}
 
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ env.GITHUB_TOKEN }}
+
+      - uses: elastic/apm-pipeline-library/.github/actions/docker-login@main
+        with:
+          registry: docker.elastic.co
+          secret: secret/observability-team/ci/docker-registry/prod
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+
       - name: Configure git
```

Comment:
I'm not 100% sure, tt's used later on when pushing the current tag. I'd say to keep it for the time being so we will revisit this release automation when we don't need maven

Reply:
If I remember correctly, you did this because the credentials were persisted and you needed to find a way to set the new credentials.

using `with.token` in the checkout action already solves this.

Reply:
OMG, I recall something but not much :/